Make polygon contracts node work again

- [x] There are some issues with the polygon node:

* Errors:
```
2022-05-26T06:38:41.886Z [ERROR] polygon.blockchain: failed to read body: err="not found"
2022-05-26T06:38:41.889Z [ERROR] polygon.blockchain: failed to read body: err="not found"
2022-05-26T06:38:41.893Z [ERROR] polygon.blockchain: failed to read body: err="not found"
2022-05-26T06:38:41.896Z [ERROR] polygon.blockchain: failed to read body: err="not found"
```
* Unable to run CI tests into GH

- [x] Maybe upgrade the version of the polygon node

--- a/keeper-polygon-localnet.Dockerfile
+++ b/keeper-polygon-localnet.Dockerfile
@@ -37,7 +37,7 @@ ENV KEEPER_RPC_PORT=8545
 
 RUN /nevermined-contracts/scripts/keeper_deploy_polygon_dockerfile.sh
 
-FROM 0xpolygon/polygon-sdk:0.1.0
+FROM 0xpolygon/polygon-edge:0.4.1
 LABEL maintainer="Nevermined <root@nevermined.io>"
 
 COPY scripts/keeper_entrypoint_polygon.sh /

--- a/keeper-polygon-localnet.Dockerfile
+++ b/keeper-polygon-localnet.Dockerfile
@@ -1,8 +1,8 @@
-FROM 0xpolygon/polygon-sdk:0.1.0 as polygon
+FROM 0xpolygon/polygon-edge:0.4.1 as polygon
 
 FROM node:14-alpine as deploy
 
-COPY --from=polygon /usr/local/bin/polygon-sdk /usr/local/bin/polygon-sdk
+COPY --from=polygon /usr/local/bin/polygon-edge /usr/local/bin/polygon-edge
 
 RUN apk add --no-cache --update\
       bash\

--- a/scripts/keeper_deploy_polygon_dockerfile.sh
+++ b/scripts/keeper_deploy_polygon_dockerfile.sh
@@ -11,7 +11,7 @@ if [ "${DEPLOY_CONTRACTS}" = "true" ]
 then
     cd /polygon-sdk
     rm -rf test-chain
-    polygon-sdk server --dev --chain genesis.json &
+    polygon-edge server --dev --chain genesis.json --data-dir test-chain &
 
     until curl --data '{"method":"web3_clientVersion","params":[],"id":1,"jsonrpc":"2.0"}' -H "Content-Type: application/json" -X POST localhost:8545
     do

--- a/scripts/keeper_deploy_polygon_dockerfile.sh
+++ b/scripts/keeper_deploy_polygon_dockerfile.sh
@@ -32,6 +32,9 @@ then
     touch /nevermined-contracts/artifacts/ready
 fi
 
+# more gracefully shutdown the node
+pkill polygon-edge
+
 # Fix file permissions
 EXECUTION_UID=$(id -u)
 EXECUTION_GID=$(id -g)

--- a/scripts/keeper_entrypoint_polygon.sh
+++ b/scripts/keeper_entrypoint_polygon.sh
@@ -9,4 +9,4 @@ cp -rp /artifacts/* /nevermined-contracts/artifacts/
 cp -rp /circuits/* /nevermined-contracts/circuits/
 
 cd /polygon-sdk
-exec polygon-sdk server "$@"
+exec polygon-edge server "$@"

